### PR TITLE
Implement libsodium Crypter

### PR DESCRIPTION
**Pull request by IrinaSed**

### CODE REVIEW

File: link_libsodium.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+test -d tmp/libsodium || mkdir -p tmp/libsodium
+
+cd tmp/libsodium
+
+curl -sL https://download.libsodium.org/libsodium/releases/$LIBSODIUM_VERSION.tar.gz -o $LIBSODIUM_VERSION.tar.gz
+tar xfz $LIBSODIUM_VERSION.tar.gz --strip-components=1
+
+./configure --prefix $PWD
+make && make check && make install
```
Comment (g0djan) [inline link_libsodium.sh]: Don't invoke make commands from the script which is invoked from Makefile.

File: internal/crypto/libsodium/crypter.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
+package libsodium
+
+// #cgo CFLAGS: -I../../../tmp/libsodium/include
+// #cgo LDFLAGS: -L../../../tmp/libsodium/lib -lsodium
+// #include <sodium.h>
```
Comment (g0djan) [inline internal/crypto/libsodium/crypter.go]: Delete commented lines

File: internal/crypto/libsodium/crypter_test.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+func TestMockCrypterFromKey_ShouldReturnErrorOnEmptyKey(t *testing.T) {
+	assert.Error(t, CrypterFromKey("").(*Crypter).setup(), "no error on empty key")
+}
+
+func TestMockCrypterFromKeyPath_ShouldReturnErrorOnNonExistentFile(t *testing.T) {
+	assert.Error(t, CrypterFromKeyPath("").(*Crypter).setup(), "no error on non-existent key path")
+}
+
+func EncryptionCycle(t *testing.T, crypter crypto.Crypter) {
+	secret := strings.Repeat(" so very secret thingy ", 1000)
```
Comment (g0djan) [inline internal/crypto/libsodium/crypter_test.go]: thing

File: internal/crypto/libsodium/crypter.go (hunk 2 above)
Comment (g0djan) [reply internal/crypto/libsodium/crypter.go]: Or is it for "C" module?

Comment (IrinaSed) [reply internal/crypto/libsodium/crypter.go]: Yes, this is special comments for cgo.

File: link_libsodium.sh (hunk 1 above)
Comment (IrinaSed) [reply link_libsodium.sh]: This is necessary to install the [libsodium](https://libsodium.gitbook.io/doc/installation#compilation-on-unix-like-systems).

**Comment by g0djan** [comment]
Make special target to build with libsodium

File: internal/configure_crypter.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// +build !libsodium
+
+package internal
+
+import (
+	"github.com/wal-g/wal-g/internal/crypto"
+)
+
```
Comment (g0djan) [inline internal/configure_crypter.go]: It would be better to add some explanation comment

Comment (g0djan) [reply internal/configure_crypter.go]: All other things look pretty good

**Pull request merged**